src/__tests__/metatests.js

```diff
@@ -73,3 +74,115 @@ describe('jsdom-global', () => {
     }
   });
 });
+
+describe('lolex', () => {
+  const lolex: Lolex = new Lolex();
+
+  afterAll(() => {
+    lolex.dispose();
+  });
+
+  afterEach(() => {
+    // clear any unset timers
+    lolex.clearAllTimers();
+  });
+
+  test('Date.now() is mocked by Lolex', () => {
+    const start = Date.now();
+    lolex.advanceTimersByTime(5.5e9); // ~63.66 days
+    const end = Date.now();
+
+    const apparentDuration = end - start;
+
+    expect(apparentDuration).toBeGreaterThan(5e9);
+    expect(apparentDuration).toBeLessThan(6e9);
+  });
+
+  test('setInterval is triggered by Lolex', () => {
+    let count = 0;
+    setInterval(() => {
+      ++count;
+    }, 1e9);
+
+    lolex.advanceTimersByTime(5.5e9);
+
+    expect(count).toBe(5);
+  });
+
+  test('setTimeout is triggered by Lolex', () => {
+    let flag = false;
+
+    setTimeout(() => {
+      flag = true;
+    }, 3e6);
+    setTimeout(() => {
+      flag = false;
+    }, 6e6);
+
+    lolex.advanceTimersByTime(4.5e6);
+
+    expect(flag).toBe(true);
+  });
+
+  test('timer count is properly maintained', () => {
+    expect(lolex.getTimerCount()).toBe(0);
+    setInterval(() => {}, 10);
+    setInterval(() => {}, 10);
+    setInterval(() => {}, 10);
+    setInterval(() => {}, 10);
+    setInterval(() => {}, 10);
+
+    expect(lolex.getTimerCount()).toBe(5);
+
+    setTimeout(() => {}, 10);
+    setTimeout(() => {}, 10);
+    setTimeout(() => {}, 10);
+    setTimeout(() => {}, 10);
+    setTimeout(() => {}, 10);
+
+    expect(lolex.getTimerCount()).toBe(10);
+
+    lolex.advanceTimersByTime(20);
+    expect(lolex.getTimerCount()).toBe(5);
+
+    lolex.clearAllTimers();
+    expect(lolex.getTimerCount()).toBe(0);
+  });
+
+  test('runOnlyPendingTimers runs timers as expected', () => {
+    const start = Date.now();
+
+    // set up eternal interval
+    let count = 0;
+    const interval = 10;
+    setInterval(() => ++count, interval);
+
+    const maxTime = 1000;
+
+    // set up several timeouts, of length at most `maxTime`
+    const timeOuts = 37;
+    for (const k of Array(timeOuts).keys()) {
+      setTimeout(() => {}, ((k + 1) / timeOuts) * maxTime);
+    }
+
+    lolex.runOnlyPendingTimers();
+    const end = Date.now();
+
+    // A long time has passed...
+    expect(end - start).toBeGreaterThanOrEqual(maxTime);
+    // ... only the interval timer is still active...
+    expect(lolex.getTimerCount()).toBe(1);
+    // ... and it has been fired appropriately many times.
+    expect(count).toBeCloseTo((end - start) / interval, 1);
+  });
+
+  test('timers fail on overflow', () => {
+    expect(() => {
+      setTimeout(() => {}, 10e10);
+    }).toThrow();
+
+    expect(() => {
+      setInterval(() => {}, 10e10);
+    }).toThrow();
+  });
```

Comment:
Why 10e10 in these tests and not something close to the required threshold, like 3e9 or 2**31?

Reply:
I like 10e10 more than 3e9. That's about it, really.

It's probably worth adding tests that are just barely on either side of the limit, as well as versions with `NaN` and `Infinity`.